Lib/test/test_structseq.py

```diff
@@ -122,5 +127,53 @@ def test_extended_getslice(self):
                     self.assertEqual(list(t[start:stop:step]),
                                      L[start:stop:step])
 
+    @support.cpython_only
+    @unittest.skipIf(ctypes is None, 'need ctypes')
+    def test_new_invalid_type(self):
+        # bpo-31573: PyStructSequence_New() expects a structseq type
+        PyStructSequence_New = ctypes.pythonapi.PyStructSequence_New
+        PyStructSequence_New.argtypes = (ctypes.py_object,)
+        PyStructSequence_New.restype = ctypes.py_object
+
+        with self.assertRaises(TypeError):
+            PyStructSequence_New(None)
+
+        class MissingNFields:
```

Comment:
Add `n_sequence_fields` in all classes except `MissingNSequenceFields`. Otherwise the test depends on the order of checks.

Shouldn't all these classes be `tuple` subclasses?